src/main/java/com/exasol/sql/ddl/drop/DropSchema.java

```diff
@@ -0,0 +1,104 @@
+package com.exasol.sql.ddl.drop;
+
+import com.exasol.sql.AbstractFragment;
+import com.exasol.sql.SqlStatement;
+import com.exasol.sql.ddl.Schema;
+
+/**
+ * This class implements an SQL {@link DropSchema} statement
+ */
+public class DropSchema extends AbstractFragment implements SqlStatement, DropSchemaFragment {
+    private Schema schema;
+    private Cascade cascade;
+    private Restrict restrict;
+    private boolean ifExists = false;
+
+    /**
+     * Create a new instance of an {@link DropSchema} statement
+     *
+     * @param schemaName name of the table to drop
+     */
+    public DropSchema(final String schemaName) {
+        super(null);
+        this.schema = new Schema(this, schemaName);
+    }
+
+    /**
+     * Add IF EXISTS clause into a DROP SCHEMA statement
+     *
+     * @return <code>this</code> for fluent programming
+     */
+    public synchronized DropSchema ifExists() {
+        if (!this.ifExists) {
+            this.ifExists = true;
+        }
+        return this;
+    }
+
+    /**
+     * Add CASCADE clause into a DROP SCHEMA statement
+     *
+     * @return <code>this</code> for fluent programming
+     */
+    public synchronized DropSchema cascade() {
+        cascade = new Cascade(this);
```

Comment:
Letting both `cascade()` and `restrict()` return a `DropSchema` allows to use the mutually exclusive options together at compile time.

There are possible alternatives:

1. Introduce a "terminator" class and return an object of the class that prevents further chaining. In this case the `restrict()` and `cascade()` must be the last methods in a chain.
1. Introduce two separate classes that are given back. While this is convenient for the end users, it creates a lot of duplication and does not scale if the chain gets more complex.
1. Return `void`

Reply:
fixed